python/ray/tests/conftest.py

```diff
@@ -125,10 +126,19 @@ def _ray_start(**kwargs):
     init_kwargs.update(kwargs)
     # Start the Ray processes.
     address_info = ray.init(**init_kwargs)
+    agent_pids = []
```

Comment:
I am confused why we need this? It should be automatically done right?

Reply:
The main reason here is that the previous test did not kill the agent process immediately, In the worst case, the agent process will exit 2s after the raylet exits. At this time, the agent process of the latter test has been started, and the HTTP port is found to be occupied by the previous agent, which will cause flaky test.

Reply:
detail:
https://github.com/ray-project/ray/blob/67c0a69643ba92333b19a5383583d001774cfab9/dashboard/agent.py#L227

Reply:
and agent process not in `Node.all_processes`, so it seems we can't kill agent process direct on `ray.shutdown`